<launch>
    <arg name="depth_image" default="zed2i/zed_node/depth/depth_registered"/>
    <arg name="depth_info" default="zed2i/zed_node/depth/camera_info"/>

    <arg name="mavros_obstacle_topic" default="/mavros/obstacle/send"/>

    <node pkg="depth_image_to_mavlink" type="depth_image_to_mavlink_node" respawn="false" name="depth_image_to_mavlink" output="screen">
        <param name="depth_image" value="$(arg depth_image)"/>
        <param name="depth_info" value="$(arg depth_info)"/>
        <param name="mavros_obstacle_topic" value="$(arg mavros_obstacle_topic)"/>
    </node>

    <arg name="use_rviz" default="false"/>

    <group if="$(arg use_rviz)">
        <node type="rviz" name="depth_rviz" pkg="rviz" args="-d $(find depth_image_to_mavlink)/config/depth_smoothed.rviz" />
    </group>
</launch>